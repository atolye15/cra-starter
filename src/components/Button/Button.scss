@import '../../styles/index';

.Button {
  padding: 16px 32px;

  &-primary {
    background-color: $color-primary;
    color: $white-color;
  }

  &-secondary {
    background-color: $color-secondary;
    color: $black-color;
  }
}
