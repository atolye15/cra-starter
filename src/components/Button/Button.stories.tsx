import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from './Button';

storiesOf('Button', module)
  .add('Primary', () => <Button primary>Primary Button</Button>)
  .add('Secondary', () => <Button secondary>Secondary Button</Button>);
