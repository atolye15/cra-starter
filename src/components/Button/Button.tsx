import React, { FunctionComponent } from 'react';
import cx from 'classnames';

import './Button.scss';

type Props = JSX.IntrinsicElements['button'] & {
  primary?: boolean;
  secondary?: boolean;
};

const Button: FunctionComponent<Props> = ({ primary, secondary, className, children, ...rest }) => {
  const classNames = cx(
    'Button',
    { 'Button-primary': primary, 'Button-secondary': secondary },
    className,
  );

  return (
    <button className={classNames} {...rest}>
      {children}
    </button>
  );
};

export default Button;
