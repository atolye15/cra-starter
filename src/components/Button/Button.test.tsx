import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

it('renders correctly primary button', () => {
  const wrapper = shallow(<Button primary>Primary Button</Button>);

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly secondary button', () => {
  const wrapper = shallow(<Button secondary>Secondary Button</Button>);

  expect(wrapper).toMatchSnapshot();
});
