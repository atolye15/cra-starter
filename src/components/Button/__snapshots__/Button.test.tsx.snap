// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly primary button 1`] = `
<button
  className="Button Button-primary"
>
  Primary Button
</button>
`;

exports[`renders correctly secondary button 1`] = `
<button
  className="Button Button-secondary"
>
  Secondary Button
</button>
`;
